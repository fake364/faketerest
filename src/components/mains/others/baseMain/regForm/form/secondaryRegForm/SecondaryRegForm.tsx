import React from 'react';
import Link from 'next/link';
import { useTranslation } from 'next-i18next';

type Props = { onTriggerFormMode?: () => void; isRegistrationForm: boolean };

const SecondaryRegForm: React.FC<Props> = ({
  onTriggerFormMode,
  isRegistrationForm
}) => {
  const { t } = useTranslation('main-page');

  return (
    <>
      <div className="mt-[14px] text-center text-[12px] text-gray-600">
        {t('regForm.form.terms.youAgreeWith')}{' '}
        <span className="font-bold whitespace-nowrap">
          <Link href={'/tos'}>{t('regForm.form.terms.termsOfService')}</Link>
        </span>{' '}
        {t('regForm.form.terms.youVeRead')}{' '}
        <span className="font-bold whitespace-nowrap">
          <Link href={'/privacy'}>{t('regForm.form.terms.privacyPolicy')}</Link>
        </span>
      </div>
      {onTriggerFormMode && (
        <div
          className="mt-[12px] text-[13px] text-center text-gray-600 cursor-pointer font-semibold"
          onClick={onTriggerFormMode}
        >
          {!isRegistrationForm
            ? t('regForm.form.registerLink')
            : t('regForm.form.logInLink')}
        </div>
      )}
      <div className=" mt-[10px] text-[13px] text-center text-gray-600 font-semibold pb-[14px]">
        {t('regForm.form.areYouABusiness')}{' '}
        <span>
          <Link href={'/business/create'}>
            {t('regForm.form.getStartedHere')}
          </Link>
        </span>
      </div>
    </>
  );
};

export default SecondaryRegForm;
