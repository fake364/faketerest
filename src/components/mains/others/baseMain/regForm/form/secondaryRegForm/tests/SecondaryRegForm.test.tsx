import React from 'react';
import { shallow } from 'enzyme';
import SecondaryRegForm from '../SecondaryRegForm';

describe('SecondaryRegForm', () => {
  it('should render switcher if there is a trigger callback', () => {
    const wrapper = shallow(
      <SecondaryRegForm
        isRegistrationForm={false}
        onTriggerFormMode={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render registration button', () => {
    const wrapper = shallow(
      <SecondaryRegForm isRegistrationForm={true} onTriggerFormMode={jest.fn} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render switcher if registration trigger is not present', () => {
    const wrapper = shallow(<SecondaryRegForm isRegistrationForm={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
