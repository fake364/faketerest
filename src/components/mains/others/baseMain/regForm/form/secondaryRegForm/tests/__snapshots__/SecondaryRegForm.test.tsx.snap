// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryRegForm should not render switcher if registration trigger is not present 1`] = `
<Fragment>
  <div
    className="mt-[14px] text-center text-[12px] text-gray-600"
  >
    regForm.form.terms.youAgreeWith
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/tos"
      >
        regForm.form.terms.termsOfService
      </ForwardRef(LinkComponent)>
    </span>
     
    regForm.form.terms.youVeRead
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/privacy"
      >
        regForm.form.terms.privacyPolicy
      </ForwardRef(LinkComponent)>
    </span>
  </div>
  <div
    className=" mt-[10px] text-[13px] text-center text-gray-600 font-semibold pb-[14px]"
  >
    regForm.form.areYouABusiness
     
    <span>
      <ForwardRef(LinkComponent)
        href="/business/create"
      >
        regForm.form.getStartedHere
      </ForwardRef(LinkComponent)>
    </span>
  </div>
</Fragment>
`;

exports[`SecondaryRegForm should render registration button 1`] = `
<Fragment>
  <div
    className="mt-[14px] text-center text-[12px] text-gray-600"
  >
    regForm.form.terms.youAgreeWith
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/tos"
      >
        regForm.form.terms.termsOfService
      </ForwardRef(LinkComponent)>
    </span>
     
    regForm.form.terms.youVeRead
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/privacy"
      >
        regForm.form.terms.privacyPolicy
      </ForwardRef(LinkComponent)>
    </span>
  </div>
  <div
    className="mt-[12px] text-[13px] text-center text-gray-600 cursor-pointer font-semibold"
    onClick={[Function]}
  >
    regForm.form.logInLink
  </div>
  <div
    className=" mt-[10px] text-[13px] text-center text-gray-600 font-semibold pb-[14px]"
  >
    regForm.form.areYouABusiness
     
    <span>
      <ForwardRef(LinkComponent)
        href="/business/create"
      >
        regForm.form.getStartedHere
      </ForwardRef(LinkComponent)>
    </span>
  </div>
</Fragment>
`;

exports[`SecondaryRegForm should render switcher if there is a trigger callback 1`] = `
<Fragment>
  <div
    className="mt-[14px] text-center text-[12px] text-gray-600"
  >
    regForm.form.terms.youAgreeWith
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/tos"
      >
        regForm.form.terms.termsOfService
      </ForwardRef(LinkComponent)>
    </span>
     
    regForm.form.terms.youVeRead
     
    <span
      className="font-bold whitespace-nowrap"
    >
      <ForwardRef(LinkComponent)
        href="/privacy"
      >
        regForm.form.terms.privacyPolicy
      </ForwardRef(LinkComponent)>
    </span>
  </div>
  <div
    className="mt-[12px] text-[13px] text-center text-gray-600 cursor-pointer font-semibold"
    onClick={[MockFunction]}
  >
    regForm.form.registerLink
  </div>
  <div
    className=" mt-[10px] text-[13px] text-center text-gray-600 font-semibold pb-[14px]"
  >
    regForm.form.areYouABusiness
     
    <span>
      <ForwardRef(LinkComponent)
        href="/business/create"
      >
        regForm.form.getStartedHere
      </ForwardRef(LinkComponent)>
    </span>
  </div>
</Fragment>
`;
